components/browser/session/src/main/java/mozilla/components/browser/session/engine/EngineObserver.kt

```diff
@@ -207,7 +207,7 @@ internal class EngineObserver(
             0,
             INITIATED,
             userAgent,
-            Environment.DIRECTORY_DOWNLOADS
+            pathFromSettings
```

Comment:
We could add an additional parameter to `onExternalResource`  were we can pass `pathFromSettings` from the engine see :

https://github.com/mozilla-mobile/android-components/blob/16343bb7702b39a9ec04ca64e5cb2314d580b85a/components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt#L758